src/mlpack/core/optimizers/sa/sa.hpp

```diff
@@ -193,8 +172,11 @@ class SA
    * @param sweepCounter Current counter representing how many sweeps have been
    *      completed.
    */
-  void GenerateMove(arma::mat& iterate,
+  template<typename FunctionType>
+  void GenerateMove(FunctionType& function,
+                    arma::mat& iterate,
                     arma::mat& accept,
+                    arma::mat& moveSize,
```

Comment:
Do you mind to add a comment about ``moveSize``, don't feel obligated we can do this later.